ledger/src/blockstore_processor.rs

```diff
@@ -189,16 +203,18 @@ fn execute_batch(
         .is_active(&feature_set::gate_large_block::id())
     {
         let execution_cost_units = aggregate_total_execution_units(timings) - pre_process_units;
+        let additional_cost_units = sum_additional_costs(batch);
         let remaining_block_cost_cap = cost_capacity_meter
             .write()
             .unwrap()
-            .accumulate(execution_cost_units);
+            .accumulate(execution_cost_units + additional_cost_units);
 
         debug!(
```

Comment:
Do you see any block being dropped by this gating feature during tests? I'm very interested to see if block produced by leader would be considered as too-large by validators. Such case should be carefully avoid.

Reply:
Not in normal tests, but when I made the cap lower (by hacking the code), I could make tests fail, and I added a few print! statements to show that valid numbers were coming through in the additional_cost section of this.